src/components/Public/ImportRecords.vue

```diff
@@ -0,0 +1,173 @@
+              @click="selectAll(true)">{{ $t('field.selector.selectAll') }}</b-button>
+          <b-button
+              class="ml-2"
+              variant="secondary"
+              @click="selectAll(false)">{{ $t('field.selector.unselectAll') }}</b-button>
+        </div>
+        <div class="row m-0">
+          <div v-for="(c, index) in importColumns" :key="index" class="form-check col-4">
+            <label class="form-check-label" :for="index">
+              <input type="checkbox" :true-value="true" :false-value="false" class="form-check-input"  v-model="c.import" :id="index">
+                {{ c.name }}
+            </label>
+          </div>
+        </div>
+      </label>
+      <div slot="modal-footer">
+          <b-button
+              :disabled="!importColumns.filter(c => c.import).length > 0"
+              variant="primary"
+              @click="importRecords()">{{ $t('general.label.import') }}</b-button>
+      </div>
+    </b-modal>
+  </form>
+</template>
+
+<script>
+import triggerRunner from '@/mixins/trigger_runner'
+import Record from '@/lib/record'
+
+export default {
+  mixins: [
+    triggerRunner,
+  ],
+
+  props: {
+    module: {
+      type: Object,
+      required: true,
+    },
+  },
+
+  data () {
+    return {
+      importColumns: [],
+      records: null,
+      importErr: null,
+      processing: false,
+      show: false,
+    }
+  },
+
+  methods: {
+    importRecords () {
+      this.processing = true
+      try {
+        this.records.forEach(async r => {
+          let record = new Record(this.module, {})
+          for (let p of this.importColumns) {
+            if (p.import) {
+              record.values[p.name] = r[p.name]
+            }
+          }
+          await this.$compose.recordCreate(record).then(nr => {
+            console.log(nr)
+          })
+        })
+        this.raiseSuccessAlert(this.$t('notification.import.successful'))
+      } catch (e) {
+        this.raiseWarningAlert(this.$t(`notification.import.failed`))
+      }
+      this.cancelImport()
+    },
+
+    getModuleID (moduleName) {
+      // find module, replaceID
+      const matchedModules = this.modules.filter(m => m.name === moduleName)
+      if (matchedModules.length > 0) {
+        return matchedModules[0].moduleID
+      } else {
+        return null
+      }
+    },
+
+    selectAll (selectAll) {
+      this.importColumns = this.importColumns.map(i => {
+        i.import = selectAll && true
+        return i
+      })
+    },
+
+    cancelImport () {
+      this.show = false
+      this.records = null
+      this.processing = false
+      this.$refs.importForm.reset()
+    },
+
+    loadFile (e) {
+      const file = e.target.files[0]
+      if (file) {
+        this.processing = true
+        const ext = file.name.split('.').pop()
+        const reader = new FileReader()
+        reader.readAsText(file)
+        reader.onload = (evt) => {
+          try {
+            if (ext === 'csv') {
```

Comment:
Can you split up these 2 importers (csv, jsonl) in their own functions? Mabie even put them in a new `lib/importer` file.